iroh-net/src/magicsock.rs

```diff
@@ -1821,6 +1821,7 @@ impl Actor {
                         warn!(?node_addr, "unable to add discovered node address to the node map: {e:?}");
                     }
                 }
+                // This case will never hit. TODO: Figure out what the intention was here
```

Comment:
I think the intention was to capture when the branches above would respond with `None`/`Err`, and at least log it to notice it

Reply:
Presumably we don't want the actor to exit in those cases so we need to add a bunch of bools to prevent tokio from polling those branches once they return `None`. Otherwise we'll basically end up with a busy wait of getting `None` -> logging and `continue`ing -> repeat.